presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -156,15 +157,23 @@ protected RelationPlan visitTable(Table node, Void context)
 
         ImmutableList.Builder<Symbol> outputSymbolsBuilder = ImmutableList.builder();
         ImmutableMap.Builder<Symbol, ColumnHandle> columns = ImmutableMap.builder();
+        Optional<Symbol> isValidColumn = Optional.empty();
```

Comment:
This in not very clean, I called this out in the doc too. Best would have been Hive connector creating FilterNode when connectors are involved in planning but right now this seemed to be the simplest option. The logic is:
1. From the record reader, generate $isValid synthetic column for each row of the page by cross referencing it with list of deleted rows from DELETE_DELTAS
2. In FilterNode, filter out rows which have this set to false.
I went with this approach of using FilterNode to filter rows rather than record reader doing it so that we can re-use the lazy materialization logic.

Reply:
Updated with the approach of using DictionaryBlock which doesnt need changes in main